### PR TITLE
add --with-configure option

### PR DESCRIPTION
**Pull request by zebardy**
As suggested in issue https://github.com/miyagawa/cpanminus/issues/398, this is a patch to add in support for --with-configure and --without-configure

### CODE REVIEW

File: lib/App/cpanminus/script.pm [hunk 1]
```diff
@@ -91,6 +91,7 @@ sub new {
         installed_dists => 0,
         install_types => ['requires'],
         with_develop => 0,
+        with_configure => undef,
```
Comment (miyagawa) [inline lib/App/cpanminus/script.pm]: Maybe use '0' for consistency?

**Comment by miyagawa** [comment]
great work! There seems to be a merge conflict, can you rebase?

File: lib/App/cpanminus/script.pm (hunk 1 above)
Comment (zebardy) [inline lib/App/cpanminus/script.pm]: That was my original approach. However the current behaviour, is if nothing is set then configure dependencies should be installed when dependencies are read from META files, but not when read from cpanfile. This required 3 possible states:-

True : install configure dependencies
False: do not install configure dependencies
undef: install configure dependencies when dependencies are read from META files, but not when read from cpanfile

I'm happy to amend things however.

Comment (zebardy) [inline lib/App/cpanminus/script.pm]: Having pull request together and thinking this logic through again has made me spot a mistake in the logic for when installing configure dependencies when reading the dependencies from META files. Fix incomming...

Comment (zebardy) [inline lib/App/cpanminus/script.pm]: Other approaches to this that come to mind are 

1) split up the state of with-configure and without-configure to be 2 separate booleans

2) remove the --without-configure option (mostly there to continue the pattern from --with-develop/--without-develop)

3) make cpanm's default behaviour consistent when reading dependencies from META files or a cpanfile  

Each approach has it's relative merits, however I felt the current approach followed the pattern from --with-develop/--without-develop, maintaining the current default behaviour in the simplest way. I'm happy to rework things however if you have a different preference or suggestion?

Comment (miyagawa) [inline lib/App/cpanminus/script.pm]: I think this is a bit complicated, and I would add `--with-configure` to enable configure when reading from cpanfile, `--without-configure` to disable that, which is default. And do not change the behavior for META files.

Comment (zebardy) [inline lib/App/cpanminus/script.pm]: Done :)

**Comment by miyagawa** [comment]
:+1:

**Pull request merged**